Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2222,7 +2249,36 @@ ExceptionOr<Ref<ImageData>> CanvasRenderingContext2DBase::getImageData(int sx, i
 
     ASSERT(pixelBuffer->format().colorSpace == toDestinationColorSpace(computedColorSpace));
 
-    return { { ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull())) } };
+    auto imageData = ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull()));
+    
+#if ENABLE(NOISY_APIS)
+    // auto identifier = IdentifierProvider::provider().identifier();
+    // (void)identifier;
+    // FIXME: Use identifier...
+
+    auto& data = imageData->pixelBuffer()->data();
+    // WTFLogAlways("getImageData: STARTING");
+    // WTFReportBacktrace();
+    for (size_t i = 0; i < data.length(); i += 4) {
+        if (randomNumber() > 0.75) {
+            int fivePercentOffset = static_cast<int>(randomNumber() * 26 - 13);
+            // WTFLogAlways("Random offset: %d", five_percent_offset);
+            size_t channel = static_cast<size_t>(randomNumber() * 3);
+            // WTFLogAlways("Random channel: %zu", channel);
+            
+            if (data.item(i + channel) + fivePercentOffset > 255)
+                data.set(i + channel, 255);
+            else if (data.item(i + channel) + fivePercentOffset < 0)
+                data.set(i + channel, 0);
+            else
+                data.set(i + channel, (data.item(i + channel) + fivePercentOffset));
```

Comment:
We can store `i + channel` and `(data.item(i + channel) + fivePercentOffset)` in a variable for readability and efficiency.

Reply:
This can also be rewritten to use `std::clamp`.

Reply:
Stored `i + channel` for now. Will look into `std::clamp`, thanks.